### PR TITLE
Docs update

### PR DESCRIPTION
**Pull request by QuzarDC**
This is a set of refreshes made by @darcagn to the documentation over here: https://github.com/darcagn/KallistiOS/tree/docs-update . Welcoming any other review. I had gone through them some time ago and did not have any objections. A majority of the changes were to move us towards more friendly and modern markdown formatted files for most, but also there was some polishing of individual blocks of text, streamlining of others, and the addition of some new md files to subfolders that allows for nicer browsing through the repo on git.

Looking at the diffs is kind of difficult to see the impact of the changes, so I'd suggest browsing the branch on github, as then it's possible to easily see the visual refresh this provides especially for the readme files that get picked up to be automatically displayed.

Aside from the changes directly from darc's repo, the following were updated:
- CHANGELOG updated to reflect the newest changes, as well as various changes from darc that hadn't been documented.
- License info updated to remove outdated references to lwip and to remove the different callouts to licenses for the kos-ports and the idea of GPL/LGPL/BSD licensed pieces of mainline KOS. At the time the docs were originally written, KOS still contained such code but it has all since been replaced. Additionally kos-ports has been completely rewritten and now consists of dozens of libraries rather than the few that were being mentioned explicitly in the documentation. Finally, we no longer use lwip at all.
- A summary was added to the RELNOTES specifically for the new functionality in v2.1.0 as there is so much.
- Replaced placeholder text with short descriptions in README.md

### CODE REVIEW

**Comment by QuzarDC** [comment]
Got one more block of changes sent over to incorporate. If any further things are needed they'll go into a subsequent PR rather than moving the goal posts here.

File: doc/README.md [hunk 1]
```diff
@@ -52,7 +27,7 @@ What KallistiOS provides optionally:
 
 What KallistiOS is NOT:
 - UNIX (or even compatible)
-- Linux/BSD (those are their own ports! =)
+- Linux/BSD (those are their own ports! :smiley:
```
Comment (andressbarajas) [inline doc/README.md:30]:
```suggestion
- Linux/BSD (those are their own ports!) :smiley:
```

File: doc/README.md [hunk 2]
```diff
@@ -52,7 +27,7 @@ What KallistiOS provides optionally:
...
 - Memory protection; programs can overwrite each other at will#
```
Comment (andressbarajas) [inline doc/README.md:31]:
```suggestion
- Memory protection; programs can overwrite each other at will
```

File: doc/LICENSE.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+See [LICENSE.GPL](license/LICENSE.GPL) and [LICENSE.LGPL](license/LICENSE.LGPL)
+for the text of these licenses.
+
+## More licensing information
+If there is ever a question or you want to ask permission to do something
+else with it, please contact the authors and we'll see how we can accommodate
+you.
+
+The Free Software Foundation has a very good source of information on
+the various licenses. Please see this page if you have license compatabality
```
Comment (andressbarajas) [inline doc/LICENSE.md:83]:
```suggestion
the various licenses. Please see this page if you have license compatibility
```

File: doc/LICENSE.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+License**. For all files that are copyrighted by the authors of KallistiOS
+(Megan Potter, etc), or for any file which does not otherwise specify
+(and does not come from a third-party package like newlib), you can assume
+that this text is attached to them:
+
+> This program is free software; you can redistribute it and/or modify
+> it under the terms of the KOS License.
+>
+> This program is distributed in the hope that it will be useful,
+> but WITHOUT ANY WARRANY; without even the implied warranty of
```
Comment (andressbarajas) [inline doc/LICENSE.md:15]:
```suggestion
> but WITHOUT ANY WARRANTY; without even the implied warranty of
```

File: utils/readme.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
+# KallistiOS Utilities
+This directory contains a number of PC-side tools used for a variety of purposes. Some are meant to be used directly by users, while others are called through KallistiOS Makefiles. These utilities are built automatically when KallistiOS is built, and many KallistiOS examples depend upon them to build properly. An example of this would be using `vqenc` to generate textures from image files at build time.
+
+- [**bin2c**](bin2c/): Converts a binary file to a C integer array for inclusion in a source file
+- [**bin2o**](bin2o/): Converts a binary files an object file for linking into a project
```
Comment (andressbarajas) [inline utils/readme.md:5]:
```suggestion
- [**bin2o**](bin2o/): Converts a binary file to an object file for linking into a project
```

**Comment by andressbarajas** [comment]
I found some spelling mistakes.

File: doc/RELNOTES.md [hunk 6]
```diff
@@ -1,9 +1,56 @@
...
+provide an overview of the new functionality since v2.0.0 in 2013. We intend 
+to have more frequent versioned releases moving forward, so this kind of 
+information should be easily seen in the changelog.
+
+# Core Functionality
+* Cooperative Threading mode is no longer supported.
+* Static Thread Local Storage (TLS).
+* C11 threads and worker threads.
+* /dev/ vfs supporting null, random, and urandom.
+* VFS Expanded with readlink, rewinddir, and more compliant readdir and stat.
```
Comment (andressbarajas) [inline doc/RELNOTES.md:21]:
```suggestion
* VFS Expanded with readlink(), rewinddir(), and more compliant readdir() and stat().
```

File: doc/RELNOTES.md [hunk 7]
```diff
@@ -1,9 +1,56 @@
...
+* Expanded C language support including C11, C17, and C23.
```
Comment (gyrovorbis) [inline doc/RELNOTES.md:22]:
```suggestion
* Expanded C language support including C11, C17, and C23.
* Expanded C++ language support including C++11, C++14, C++17, C++20, C++23, and C++26.
```

**Review by gyrovorbis — APPROVED** [review]
I'd like to at least get the C++ work and support, which was a substantial effort, mentioned, so people don't think we're C only, but other than that, good to go!

**Comment by QuzarDC** [comment]
Ok. My apologies for the back and forth, but this is now ready. The changes from the previous are relatively minor, but since it touches some of the license stuff want to make sure they get well reviewed.

**Review by Kazade — APPROVED** [review]

**Pull request merged**